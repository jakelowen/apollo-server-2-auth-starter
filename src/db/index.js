const environment = process.env.NODE_ENV || 'development';
const knexTinyLogger = require('knex-tiny-logger').default;
const config = require('../../knexfile')[environment];

// eslint-disable-next-line import/order
const db = require('knex')(config);

const { DEBUG } = process.env;
if (DEBUG) knexTinyLogger(db);

module.exports = db;
