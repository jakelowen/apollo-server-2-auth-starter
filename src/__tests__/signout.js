/* eslint-disable no-undef */
import gql from 'graphql-tag';

import { createTestClient } from 'apollo-server-testing';
import { constructTestServer, makeTestContext } from './__utils';
import db from '../db';
// import makeContext from '../utils/makeContext';
// import { makeTestContext } from './__utils';
// const nock = require('nock');

const SIGNOUT_MUTATION = gql`
    mutation {
        signout {
            code
            success
            message
        }
    }
`;


beforeAll(async () => db.migrate.latest());
beforeEach(async () => Promise.all([db.raw('TRUNCATE TABLE users CASCADE')]));
afterAll(async () => db.destroy());

describe('Signout', () => {
  test('Happy Path', async () => {
    const clearCookie = jest.fn();

    const { server } = constructTestServer({
      context: await makeTestContext(null, { res: { clearCookie }, user: { id: 1, email: 'a@a.a' } }),
    });

    const { mutate } = createTestClient(server);
    const res = await mutate({ mutation: SIGNOUT_MUTATION });
    // console.log(res);
    expect(clearCookie).toHaveBeenCalled();
    expect(clearCookie).toBeCalledWith('token');
    expect(res).toMatchSnapshot();
  });

  test('throws error if not logged in', async () => {
    const clearCookie = jest.fn();
    const { server } = constructTestServer({
      context: await makeTestContext({ res: { clearCookie } }),
    });
    const { mutate } = createTestClient(server);
    const res = await mutate({ mutation: SIGNOUT_MUTATION });
    // console.log(res);
    expect(res.errors[0].message).toEqual('You must be logged in to do this');
  });
});
