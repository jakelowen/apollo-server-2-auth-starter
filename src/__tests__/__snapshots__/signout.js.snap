// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signout Happy Path 1`] = `
Object {
  "data": Object {
    "signout": Object {
      "code": "ok",
      "message": "Goodbye!",
      "success": true,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
